playbooks/roles/l2tp-ipsec/templates/streisand-l2tp-iptables.service.j2

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Set iptables rules required to L2TP/IPsec gateway
+After=network.target
+Before=ipsec.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+{% for rule in l2tp_ipsec_firewall_rules %}
```

Comment:
I think it might be better to place a `l2tp_ipsec_firewall_up.sh` script on the system and have it run by `ExecStart` instead of having multiple `ExecStarts` in the one service directly. What do you think?

Reply:
I don't see a value in putting iptables execution in separate script. With separate script we have potential of masking issues with some iptables call. Also the separate script doesn't help if you need to reapply iptables rules, just run `systemctl restart streisand-l2tp-iptables`.

Reply:
> With separate script we have potential of masking issues with some iptables calls

What kind of issues are you thinking? Like a non-zero exit? That's pretty easy to catch with `set -e`. It seems like it will cause duplication for the case where you wanted to also drop the rules no? With a script it might be easier to refactor the commonality between adding/dropping.

Why do you think it's better to have many `ExecStart`'s here? 

>  Also the separate script doesn't help if you need to reapply iptables rules, just run systemctl restart streisand-l2tp-iptables.

Wouldn't the same invocation also restart the script to reapply the firewall rules if it were the target of the `ExecStart`? If not then I agree that's a good advantage to this approach.

Reply:
> It seems like it will cause duplication for the case where you wanted to also drop the rules no With a script it might be easier to refactor the commonality between adding/dropping.

I'm not sure what kind of duplication might happen here. It's still sequential execution of iptables one way or another. Multiple ExecStop is allowed here too.
> Why do you think it's better to have many ExecStart's here?

Because this is specifically supported by `Type=oneshot`. I still don't see a value of shell script which wraps sequential execution of iptables rules while this can be done directly in systemd unit. I prefer to keep things clean without unnecessary cruft.

Reply:
For example doing `systemctl cat streisand-l2tp-iptables` allows to view what exactly service does instantly, without need to dwell deeper into wrapper shell script.

Reply:
Ok! I'm not entirely convinced but I don't want to bikeshed :-) I'm happy to leave this as-is in your PR. Thanks for engaging on the topic.